src/socket.c

```diff
@@ -32,16 +32,15 @@
 void socket_init(socket_ctx *ctx, char *path) {
 	printf("initialize socket\n");
 	ctx->socket.sun_family = AF_UNIX;
-	strcpy(ctx->socket.sun_path, path);
+	strncpy(ctx->socket.sun_path, path, strlen(path));
```

Comment:
On Linux, this is actually not an issue (and l3roamd is very specific to Linux), as either a 0 termination or the *sizeof(struct sockaddr_un)* passed to bind work to limit the string length. So while we'd truncate a path when it is longer than sun_path, we wouldn't cause any memory issues.

Other possible approaches would be to output an error when the path doesn't fit into sizeof(name.sun_path) - 1 bytes, or to use a variable-size array for the sockaddr_un, which may include more space for sun_path (which is allowed on Linux and many other implementations).